packages/integrations/image/src/lib/get-image.ts

```diff
@@ -97,24 +98,35 @@ async function resolveTransform(input: GetImageTransform): Promise<TransformOpti
 /**
  * Gets the HTML attributes required to build an `<img />` for the transformed image.
  *
- * @param loader @type {ImageService} The image service used for transforming images.
  * @param transform @type {TransformOptions} The transformations requested for the optimized image.
  * @returns @type {ImageAttributes} The HTML attributes to be included on the built `<img />` element.
  */
 export async function getImage(
-	loader: ImageService,
 	transform: GetImageTransform
 ): Promise<ImageAttributes> {
-	globalThis.astroImage.loader = loader;
+	if (!transform.src) {
+		throw new Error('[@astrojs/image] `src` is required');
+	}
+
+	let loader = globalThis.astroImage?.loader;
+
+	if (!loader) {
+		// @ts-ignore
+		const { default: mod } = await import('virtual:image-loader');
```

Comment:
Lazy importing the loader so it's only imported if/when the `getImage()` helper is used